### PR TITLE
Reduce AJAX Calls on Product Page

### PR DESCRIPTION
**Pull request by jaebird123**
- reduced the number of ajax requests on product show page by making single ajax request every 500ms from show.js.erb
- show.js.erb now only reloads partials if a task's product is not ready or a task has a test execution that is pending
- added integration test for AJAX reloading the product test links

### CODE REVIEW

File: app/helpers/products_helper.rb [hunk 1]
```diff
@@ -106,4 +106,16 @@ def with_c3_task(task)
     end
     [task]
   end
+
+  # only yeilds tasks whose product test links need to be reloaded
```
Comment (tstrass) [inline app/helpers/products_helper.rb]: yields @jaebird123

**Comment by tstrass** [comment]
I added a product with a bunch of measures and it worked through all of them, but after it finished the last 6 were stuck on "building" after the requests stopped. @jaebird123

**Comment by tstrass** [comment]
After adding a product, the bulk download progress no longer continuously updates. I think something is off with products/show.js.erb.

File: app/helpers/products_helper.rb (hunk 1 above)
Comment (jaebird123) [inline app/helpers/products_helper.rb]: ah yes, i before e except after c and those other situations

**Comment by jaebird123** [comment]
Ok, yeah I see the issue with not showing the progress of the bulk download. I'll fix that. I'll try to reproduce the measure tests stuck in `building` too and see if I can fix it

**Pull request closed without merging**